lib/solveEquation/stepThrough.js

```diff
@@ -183,11 +279,25 @@ function addSimplificationSteps(steps, equation, debug=false) {
   let oldEquation = equation.clone();
 
   const leftSteps = simplifyExpressionNode(equation.leftNode, false);
+  const simplifiedLeftNode = leftSteps.length !== 0
```

Comment:
As per our in person convo :) can you add some comments in this part of the code, explaining what you're doing, and maybe stepping through an example?